cmd/system-probe/api/server.go

```diff
@@ -33,6 +36,22 @@ func StartServer(cfg *config.Config) error {
 		return fmt.Errorf("failed to create system probe: %s", err)
 	}
 
+	// Setup telemetry server
+	port := cfg.DefaultSystemProbeExpVarPort
+	if ddconfig.Datadog.GetBool("telemetry.enabled") {
+		http.Handle("/telemetry", telemetry.Handler())
+	}
+	expvarServer := &http.Server{
+		Addr:    "127.0.0.1:" + port,
+		Handler: http.DefaultServeMux,
+	}
+	go func() {
+		err := expvarServer.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			log.Errorf("error creating expvar server on port %v: %v", port, err)
+		}
+	}()
```

Comment:
Could this new `telemetry` endpoint be added to the existing system-probe HTTP server, rather than spinning a new server on a new port dedicated just to serving prometheus telemetry?

Reply:
The existing system-probe HTTP server listens over a unix domain socket, whereas for prometheus, I believe, we need a HTTP server listening over TCP, since that is how the check seems to query the endpoint.

Reply:
The expvar metrics are also served through the debug port: https://github.com/DataDog/datadog-agent/blob/main/cmd/system-probe/app/run.go#L199

I think we should try to add the telemetry endpoint to this server.

Reply:
Done! Updated https://github.com/DataDog/k8s-resources/pull/37994 as well